.gitattributes

```diff
@@ -6,11 +9,35 @@
 ###############################################################################
 # Set default behavior for command prompt diff.
 #
-# This is need for earlier builds of msysgit that does not have it on by
-# default for csharp files.
-# Note: This is only used by command line
+# This is only used by command line
 ###############################################################################
-#*.cs     diff=csharp
+*.cs            text diff=csharp
```

Comment:
What's `diff=csharp` all about?

Reply:
The `diff` attribute sets a parser/regexp for deciding what things to include and how to format the "hunk header" which shows up when you diff on the cmd-line. More info at ["Defining a custom hunk-header"](http://schacon.github.io/git/gitattributes.html) and at the [gitattributes docs](https://www.git-scm.com/docs/gitattributes#_generating_diff_text). There are a bunch of built-in patterns, but you can also define any arbitrary pattern if you wish.